lib/lte_link_control/lte_lc.c

```diff
@@ -67,7 +67,8 @@ static char rai_param[2] = CONFIG_LTE_RAI_REQ_VALUE;
 
 /* Requested NCELLMEAS params */
 static struct lte_lc_ncellmeas_params ncellmeas_params;
-static bool ncellmeas_ongoing;
+/* Sempahore value 1 means ncellmeas is not ongoing, and 0 means it's ongoing. */
+K_SEM_DEFINE(ncellmeas_ongoing_sem, 1, 1);
```

Comment:
Just a suggestion, but the name of the semaphore would be more logical if it would be something like `ncellmeas_idle_sem`. Now it's somewhat counter-intuitive that the name is "ongoing", but value 1 means idle.

Reply:
Renamed.
I also found an error case where semaphore wasn't given and made a test for that too.